etcdserver/server.go

```diff
@@ -1018,13 +1021,21 @@ func (s *EtcdServer) apply(es []raftpb.Entry, confState *raftpb.ConfState) (uint
 				s.w.Trigger(req.ID, s.applyRequest(*req))
 			} else {
 				ar := s.applyV3Request(&raftReq)
-				s.w.Trigger(raftReq.ID, ar)
-				if ar.err == ErrNoSpace {
-					plog.Errorf("applying raft message exceeded backend quota")
-					// TODO: send alarm
-					s.errorc <- ar.err
-					return applied, true
+				if ar.err != ErrNoSpace || len(s.alarmStore.Get(pb.AlarmType_NOSPACE)) > 0 {
```

Comment:
i feel we should probably still return user the original error response rather than the alarm response at line 1037.